src/mapper.rs

```diff
@@ -0,0 +1,162 @@
+                    line_mapping,
+                    ..
+                } => {
+                    if let Some(line_mapping) = line_mapping {
+                        if let Some(original_startline) = line_mapping.original_startline {
+                            let members = class.members.entry(obfuscated).or_insert_with(|| vec![]);
+                            members.push(MemberMapping {
+                                startline: line_mapping.startline,
+                                endline: line_mapping.endline,
+                                original_class,
+                                original,
+                                original_startline,
+                                original_endline: line_mapping.original_endline,
+                            });
+                        }
+                    }
+                }
+            }
+        }
+        if !class.original.is_empty() {
+            classes.insert(class.obfuscated, class);
+        }
+
+        Self { classes }
+    }
+
+    /// Remaps a single Stackframe.
+    ///
+    /// Returns one or more [`StackFrame`]s, based on the information in
+    /// the proguard mapping. This can return more than one frame in the case
+    /// of inlined functions. In that case, frames are sorted top to bottom.
+    ///
+    /// [`StackFrame`]: struct.StackFrame.html
+    pub fn remap_frame(
+        &'s self,
+        frame: &StackFrame<'s>,
+    ) -> impl Iterator<Item = StackFrame<'s>> + 's {
+        let mut frame = frame.clone();
+        if let Some(class) = self.classes.get(frame.class.as_ref()) {
+            frame.class = class.original.into();
+            if let Some(members) = class.members.get(frame.method.as_ref()) {
+                let mut mapped = vec![];
+                for member in members {
+                    // in case we don’t find a record based on the line number,
+                    // we can still re-map the method name.
+                    if member.original_class.is_none() {
+                        frame.method = member.original.into();
+                    }
+                    // skip any members which do not match our the frames line
+                    if frame.line < member.startline || frame.line > member.endline {
+                        continue;
+                    }
+                    // parents of inlined frames don’t have an `endline`, and
+                    // the top inlined frame need to be correctly offset.
+                    let line = if member.original_endline.is_none() {
+                        member.original_startline
+                    } else {
+                        member.original_startline + frame.line - member.startline
+                    };
+                    let file = member
+                        .original_class
+                        .map(|c| {
+                            let c = c.rsplit('.').next().unwrap();
+                            Cow::Owned(format!("{}.java", c))
```

Comment:
wondering if this should be hardcoded `.java` or dynamic, a class can be `.kt` as well.
I can run some tests once it's testable with Kotlin code and see how it looks like.

Reply:
+1 on this. Let's make this an `Option` or something, and we can still have a method on `StackFrame` for convenience.